File: project-template-vite/src/main/java/com/emfabro/template/domain/entity/Card.java
package com.emfabro.template.domain.entity;


import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "cards")
@Data
public class Card {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String name;

    private String phone;

    private String email;

    private String company;

    @Column(name = "position")
    private String position;

    private String address;

    @Column(name = "style")
    private String style;

    @Column(name = "facebook")
    private Boolean facebook;

    @Column(name = "instagram")
    private Boolean instagram;

    @Column(name = "line")
    private Boolean line;

    @Column(name = "threads")
    private Boolean threads;

    @Column(name = "is_public")
    private Boolean isPublic;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: business_card_app/lib/models/card.dart
// lib/models/card.dart
class BusinessCard {
  final int? id;
  final String name;
  final String? company;
  final String? phone;
  final String? email;
  final String? address;
  final String? position;
  final String? style;
  final String? avatarUrl;
  final String? facebookUrl;
  final String? instagramUrl;
  final String? lineUrl;
  final String? threadsUrl;
  final bool? facebook;
  final bool? instagram;
  final bool? line;
  final bool? threads;
  final bool? isPublic;

  BusinessCard({
    this.id,
    required this.name,
    this.company,
    this.phone,
    this.email,
    this.address,
    this.position,
    this.style,
    this.avatarUrl,
    this.facebookUrl,
    this.instagramUrl,
    this.lineUrl,
    this.threadsUrl,
    this.facebook,
    this.instagram,
    this.line,
    this.threads,
    this.isPublic,
  });

  factory BusinessCard.fromJson(Map<String, dynamic> json) {
    return BusinessCard(
      id: json['id'],
      name: json['name'] ?? '',
      company: json['company'],
      phone: json['phone'],
      email: json['email'],
      address: json['address'],
      position: json['position'],
      style: json['style'],
      avatarUrl: json['avatarUrl'],
      facebookUrl: json['facebookUrl'],
      instagramUrl: json['instagramUrl'],
      lineUrl: json['lineUrl'],
      threadsUrl: json['threadsUrl'],
      facebook: json['facebook'],
      instagram: json['instagram'],
      line: json['line'],
      threads: json['threads'],
      isPublic: json['isPublic'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'company': company,
      'phone': phone,
      'email': email,
      'address': address,
      'position': position,
      'style': style,
      'avatarUrl': avatarUrl,
      'facebookUrl': facebookUrl,
      'instagramUrl': instagramUrl,
      'lineUrl': lineUrl,
      'threadsUrl': threadsUrl,
      'facebook': facebook,
      'instagram': instagram,
      'line': line,
      'threads': threads,
      'isPublic': isPublic,
    };
  }
}